discord/automod.py

```diff
@@ -0,0 +1,420 @@
+        it = filter(None, map(self.guild._resolve_channel, self.raw_exempt_channels))
+        return utils._unique(it)
+
+    @property
+    def actions(self) -> List[AutoModRuleAction]:
+        """List[:class:`AutoModRuleAction`]: The actions that are taken when this rule is triggered."""
+        return [AutoModRuleAction.from_data(action) for action in self._actions]
+
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = None,
+        event_type: Optional[AutoModRuleEventType] = None,
+        actions: Optional[List[AutoModRuleAction]] = None,
+        enabled: Optional[bool] = None,
+        exempt_roles: Optional[List[Role]] = None,
+        exempt_channels: Optional[List[Union[GuildChannel, Thread]]] = None,
+    ) -> Self:
+        """|coro|
+
+        This method will edit an existing auto moderation rule.
+
+        Parameters
+        -----------
+        name: Optional[:class:`str`]
+            The new name to change to.
+        event_type: Optional[:class:`AutoModRuleEventType`]
+            The new event type to change to.
+        actions: Optional[List[:class:`AutoModRuleAction`]]
+            The new rule actions to update.
+        enabled: Optional[:class:`bool`]
+            Whether the rule should be enabled or not.
+        exempt_roles: Optional[List[:class:`Role`]]
+            The new roles to exempt from the rule.
+        exempt_channels: Optional[List[Union[:class:`abc.GuildChannel`], :class:`Thread`]]
+            The new channels to exempt from the rule.
+
+        Raises
+        -------
+        :exc:`Forbidden`
+            You are not authorised to perform this action.
+        :exc:`NotFound`
+            The rule was not found.
+
+        Returns
+        --------
+        :class:`AutoModRule`
+            The updated auto moderation rule.
+        """
+        transformed_actions = None
+        if actions:
+            transformed_actions = [action.to_dict() for action in actions]
+
+        data = await self._state.http.modify_auto_moderation_rule(
+            self.guild.id,
+            self.id,
+            name=name,
+            event_type=event_type,
+            actions=transformed_actions,
+            enabled=enabled,
+            exempt_roles=exempt_roles,
+            exempt_channels=exempt_channels,
+        )
+
+        return AutoModRule(data=data, guild=self.guild, state=self._state)
+
+    async def delete(self) -> None:
+        """|coro|
+
+        This method will delete the auto moderation rule.
+
+        Raises
+        -------
+        :exc:`Forbidden`
+            You are not authorised to perform this action.
+        :exc:`NotFound`
+            The rule was not found.
+        """
+        await self._state.http.delete_auto_moderation_rule(self.guild.id, self.id)
+
+
+class AutoModRuleExecution:
+    """Represents an auto moderation rule execution.
+
```

Comment:
```suggestion
    """Represents an auto moderation rule execution.

    .. versionadded:: 2.0

```

What, exactly, is an auto moderation rule execution? I feel this needs to be described.